File: app/src/main/java/com/johnpetitto/orachat/user/UserPreferences.java
package com.johnpetitto.orachat.user;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class UserPreferences {
    private SharedPreferences preferences;

    public UserPreferences(Context context) {
        preferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public String getAuthorizationToken() {
        return preferences.getString("token", null);
    }

    @SuppressLint("ApplySharedPref")
    public void setAuthorizationToken(String token) {
        // synchronized with commit for background thread write
        preferences.edit().putString("token", token).commit();
    }
}

File: app/src/main/java/com/johnpetitto/orachat/data/user/UserPreferences.kt
package com.johnpetitto.orachat.data.user

import android.annotation.SuppressLint
import android.content.Context
import android.preference.PreferenceManager

class UserPreferences(context: Context) {
    private val preferences = PreferenceManager.getDefaultSharedPreferences(context)

    var authorizationToken: String?
        get() = preferences.getString("token", null)
        set(token) = storeToken(token)

    @SuppressLint("ApplySharedPref")
    private fun storeToken(token: String?) {
        preferences.edit().putString("token", token).commit()
    }
}
